samples/videoDecode/videodecode.cpp

```diff
@@ -329,17 +329,17 @@ int main(int argc, char **argv) {
                 std::cout << std::setfill('0') << std::setw(2) << std::hex << static_cast<int>(digest[i]);
             }
             std::cout << std::endl;
-
             if (b_md5_check) {
                 char ref_md5_string[33], c2[2];
                 uint8_t ref_md5[16];
-                std::string str;
+                std::string str(2,0);
 
                 for (int i = 0; i < 16; i++) {
                     int c;
                     ref_md5_file.get(c2[0]);
                     ref_md5_file.get(c2[1]);
-                    str = c2;
+                    str[0] = c2[0];
```

Comment:
Not sure why this needs to be changed. str = c2 OR str = str(string_name) would do
same comment for all other changes

Reply:
Because I started to see errors in this conversion but not sure why str = c2 doesn't work anymore: after the string assignment, str become a 3-letter string.

Reply:
@rrawther Added a simpler version. Please take a look.